### PR TITLE
Allow multiple parents with same reference in case details

### PR DESCRIPTION
**Pull request by proteusvacuum**
https://sentry.io/dimagi/commcarehq/issues/854613425/

When there are case types with 2 parents with the same reference id, and you reference a parent property in the case list, HQ doesn't know what to do. I introduced this in https://github.com/dimagi/commcare-hq/pull/23044
Now, it shows the property you are referencing under both parent case types. 

This bug probably only exists for eCHIS or other very complicated apps.

### CODE REVIEW

File: corehq/apps/app_manager/tests/test_case_meta.py [hunk 1]
```diff
@@ -193,3 +220,48 @@ def test_case_references_open_close(self):
         meta_type = app.get_case_metadata().get_type('save_to_case')
         self.assertEqual({}, meta_type.opened_by)
         self.assertTrue(m0f1.unique_id in meta_type.closed_by)
+
+    def test_multiple_parents_case_lists(self):
+        """If the case has multiple parents, and you reference a parent property in the
+        case list, we can't tell which parent will be shown """
+        app, _ = self.get_test_app()
+        # module 1 has case type 'child' which has a single parent, 'parent'
```
Comment (orangejenny) [inline corehq/apps/app_manager/tests/test_case_meta.py]: Appreciate the comments in tests; they can be hard to follow.

File: corehq/apps/reports/formdetails/readable.py [hunk 2]
```diff
@@ -208,32 +208,38 @@ def get_property(self, root_case_type, name):
...
+            if parent_props:
+                return parent_props
             else:
                 params = {'case_type': root_case_type, 'relationship': parent_rel}
                 raise CaseMetaException(_(
                     "Case type '%(case_type)s' has no '%(relationship)s' "
                     "relationship to any other case type.") % params)
 
-        return type_.get_property(name)
+        return [type_.get_property(name)]
```
Comment (orangejenny) [inline corehq/apps/reports/formdetails/readable.py]: Should this be renamed? I wouldn't expect `get_property` to return a list.

**Review by orangejenny — APPROVED** [review]

**Review by biyeun — APPROVED** [review]
